mobile/src/home-view.js

```diff
@@ -0,0 +1,278 @@
+ 
+    questions = sharedTasks.map(t => ({...t, type:'shared'}))
+    votes = sharedVotes.map(t => ({...t, type:'shared'}))
+    questionScore = questions.map(function(x) { return Object.assign({id: x.key, score: 0}) })
+
+    this.findScore(questionScore, votes, questions)
+    if (showRecent === true) {
+    }
+  
+ 
+   
+    return (
+      <KeyboardAvoidingView style={s.container} behavior={Platform.select({ios: "padding", android: null})}>
+        <TitleBar title="Keynote" client={client} signin={this.signin} />
+        <View style={s.compose}>
+          <TextInput style={s.composeText} placeholder="Add Question..."
+            value={this.state.question}
+            onChangeText={question => this.setState({question})} 
+            maxLength={250}
+            />
+          <View style={s.sendButtons}>
+            <TouchableOpacity style={s.sendButton} onPress={this.createSharedTask}><Text style={s.sendButtonText}>Submit 📢</Text></TouchableOpacity>
+          </View>
+        </View>
+        <View style={s.counter}>
+        <Text>{this.state.question.length}/250 </Text>
+        </View>
+        
+
+        <ScrollView style={s.scroll}>
+        
+          {questions.map(question => (
+            <View key={question.id} style={s.task}>
+              {/* { renderCreator(question) } */}
+              <Text style={s.taskText}>{question.text}</Text>
+              <TouchableOpacity onPress={() => this.newVote(question)}><Text style={s.checkmark}>👍 </Text></TouchableOpacity>
+            </View>
+          ))} 
+        </ScrollView>
+        
+       
+      </KeyboardAvoidingView>
+    )
+  }
+ 
+
+
+  findScore(questionScore, votes, questions){
+    votes.forEach(function(a) {
+    questionScore.find(function(value, index) {
+      if (value.id === a.question) {
+        value.score += a.vote
+      }
+    })
+  });
+  this.findOrder(questionScore, questions)
+  }
+
+
+  findOrder(questionScore, questions){
+  
+    questionScore.forEach(function(a){
+      questions.find(function(value, index){
+        console.log(value.creator)
+        if (value.key === a.id) {
+          value.score = a.score
+        }  
+      })
+    });
+     questions.sort(function (a,b){
+      return b.score - a.score
+    })
+  }
+
+  findOrderDate(questions){
+
+    //need to sort by date created
+
+  }
+
+  createSharedTask = () => this.createQuestion(fbc.database.public.allRef)
+ 
+  createQuestion(ref) {
+    console.log(this.state.question)
+    if (this.user && this.state.question) {
+      var time = new Date().getTime()
+      ref('questions').push({
+        text: this.state.question,
+        creator: client.currentUser.firstName + " " + client.currentUser.lastName,
+        score : 0,
+        dateCreate: time
+      })
+      .then(() => this.setState({question: ''}))
+      .catch (x => console.error(x))
+    }    
+  }
+
+  createVote(voteNum, question){
+    ref = fbc.database.public.allRef
```

Comment:
Where is `ref` declared? It doesn't look like it's used aside from 2 lines down, so might as well just say `fbc.database.public.allRef`('votes').push(...)